Documentation/contributor-guide/features.md

```diff
@@ -0,0 +1,39 @@
+# Features 
+
+This document provides general guidelines for adding and removing etcd features. The etcd features fall into two stages, experimental and stable.
+
+## Experimental
+
+Any new feature should be added as an experimental feature. An experimental feature is characterized as below:
+- Associated with an issue with a clear need for such a feature.
+- Any configuration flags related to the implementation of the feature are prefixed with `experimental` e.g. `--experimental-feature-name`.
+- Any variable names related to the implementation of the feature are prefixed with `Experimental` e.g. `ExperimentalFeatureName`.
+- Might be buggy. Enabling the feature may not work as expected. 
+- Support for such a feature may be dropped at any time without notice.
+- Disabled by default when added initially.
+
+### Graduation to Stable
+
+It is important that experimental features don't get stuck in that stage.  An experimental feature should move to the stable stage by following the lifecycle steps below:
+- A new feature added in the latest release and disabled by default e.g. etcd v3.5 or v4.0
+- Given there are no open issues against the feature, it is ready for review to,
```

Comment:
I think we should have more strict requirements for graduating feature than just no open issues. 

Things that should be considered when graduating (not all are required depending on feature) :
* Logs allowing for proper debugging
* Documentation
* Metrics
* Benchmarks
* Proper E2e tests
* upgrade/downgrade tests if there are changes in api, raft, wal protos

Graduation issue should have those requirements as a checklist and be approved by at least 2 maintainers that are willing to support this feature.